components/memory-cache/src/component.ts

```diff
@@ -25,11 +25,35 @@ export function createInMemoryCacheComponent(): ICacheStorageComponent {
     async keys(pattern?: string): Promise<string[]> {
       const allKeys = Array.from(cache.keys()) as string[]
       if (!pattern) return allKeys
-      
+
       // Simple pattern matching - convert glob-like pattern to regex
       const regexPattern = pattern.replace(/\*/g, '.*')
       const regex = new RegExp(regexPattern)
       return allKeys.filter((key: string) => regex.test(key))
+    },
+
+    async setInHash<T>(key: string, field: string, value: T, ttlInSecondsForHash?: number): Promise<void> {
+      const hash = { ...(cache.get(key) ?? {}) }
+      const options = ttlInSecondsForHash && ttlInSecondsForHash > 0 ? { ttl: ttlInSecondsForHash * 1000 } : undefined
+
+      hash[field] = value
+      cache.set(key, hash, options)
+    },
+
+    async getFromHash<T>(key: string, field: string): Promise<T | null> {
+      return cache.get(key)?.[field] ?? null
+    },
+
+    async removeFromHash(key: string, field: string): Promise<void> {
+      const hash = cache.get(key)
+      if (!hash) return
+      const newHash = { ...hash }
+      delete newHash[field]
+      cache.set(key, newHash)
```

Comment:
if `newHash` is empty, what do you thing about removing it?  
```suggestion
if (Object.keys(newHash).length === 0) {
  cache.del(key)
} else {
  cache.set(key, newHash)
}
```